dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java

```diff
@@ -234,14 +234,14 @@ private void updateMasterNodes() {
     }
 
     private void updateWorkerNodes() {
-        workerGroupWriteLock.lock();
```

Comment:
https://github.com/apache/dolphinscheduler/blob/bd48c991783b2e0ea0c602f6ef6c9a09c92e7b42/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/registry/ServerNodeManager.java#L248-L277

I guess we need to do both. `updateWorkerGroupMappings` also accesses  `workerNodeInfo`, correct?

cc @ruanwenjun

Reply:
Yes, the logic in ServerNodeManager is complex, some lock is not needed.
It's better to split the ServerNodeManager into `MasterInstanceManager`, `WorkerInstanceManager`.

And the lock here is not needed, we don't need to use a lock here, each refresh we directly replace the whole map in memory is enough, right now we use read/write lock is overdesign.